src/graphics/program-lib/programs/standard.js

```diff
@@ -1417,7 +1434,36 @@ var standard = {
         if ((lighting && options.useSpecular) || reflections) {
             code += "   getSpecularity();\n";
             if (!getGlossinessCalled) code += "   getGlossiness();\n";
-            if (options.fresnelModel > 0) code += "   getFresnel();\n";
+            if (options.fresnelModel > 0) {
+                if (!graphCode) code += "   getFresnel();\n";
+            }
+        }
+
+        if (graphCode) {
+            if (!options.normalMap) {
+                code += "   dNormalMap = vec3(0);\n";
```

Comment:
maybe should be 0.0 instead of 0, might generate error that int cannot be converted to float. Not sure.

Reply:
no error - GLSL allows this as constructors are used for type conversion